File: Android/app/src/main/java/com/hhp227/paging3/MovieListResponse.kt
package com.hhp227.paging3

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MovieListResponse(
    @SerialName("page") val page: Int,
    @SerialName("results") val results: List<Movie>,
    @SerialName("total_pages") val totalPages: Int,
    @SerialName("total_results") val totalResults: Int
)

File: iOS/Paging/MovieListResponse.swift
//
//  MovieListResponse.swift
//  Paging
//
//  Created by 홍희표 on 2022/06/19.
//

import Foundation

struct MovieListResponse: Codable, Identifiable {
    var id: Int
    
    var totalPages: Int
    
    var totalResults: Int
    
    var results: [Movie]
    
    static let EMPTY = MovieListResponse(id: 0, totalPages: 0, totalResults: 0, results: [])
    
    enum CodingKeys: String, CodingKey {
        case results
        case id = "page"
        case totalPages = "total_pages"
        case totalResults = "total_results"
    }
}
